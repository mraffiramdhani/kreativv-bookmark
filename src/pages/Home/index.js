import React from "react";
import { MdEdit, MdVisibility, MdDelete } from "react-icons/md";

function Home(props) {
  return (
    <div className="flex-grow container mx-auto sm:px-4 pt-6 pb-8">
      <div className="bg-white border-t border-b sm:border-l sm:border-r sm:rounded shadow mb-6">
        <div className="border-b px-6">
          <div className="flex justify-between -mb-px">
            <div className="text-blue-dark py-4 text-lg">Progress</div>
          </div>
        </div>
        <div className="block px-6 lg:hidden">
          <div className="w-full text-center py-8 border-b">
            <div className="text-grey-darker mb-2">
              <span className="text-5xl">0</span>
            </div>
            <div className="text-sm uppercase text-grey tracking-wide">
              Book(s) Finished
            </div>
          </div>
          <div className="w-full text-center py-8 border-b">
            <div className="text-grey-darker mb-2">
              <span className="text-5xl">0</span>
            </div>
            <div className="text-sm uppercase text-grey tracking-wide">
              Page(s) Read
            </div>
          </div>
          <div className="w-full text-center py-8">
            <div className="text-grey-darker mb-2">
              <span className="text-5xl">0</span>
              <span className="text-3xl align-top">%</span>
            </div>
            <div className="text-sm uppercase text-grey tracking-wide">
              Progress Of The Last Book
            </div>
          </div>
        </div>
        <div className="hidden lg:flex">
          <div className="w-1/3 text-center py-8">
            <div className="border-r">
              <div className="text-grey-darker mb-2">
                <span className="text-5xl">0</span>
              </div>
              <div className="text-sm uppercase text-grey tracking-wide">
                Book(s) Finished
              </div>
            </div>
          </div>
          <div className="w-1/3 text-center py-8">
            <div className="border-r">
              <div className="text-grey-darker mb-2">
                <span className="text-5xl">0</span>
              </div>
              <div className="text-sm uppercase text-grey tracking-wide">
                Page(s) Read
              </div>
            </div>
          </div>
          <div className="w-1/3 text-center py-8">
            <div>
              <div className="text-grey-darker mb-2">
                <span className="text-5xl">0</span>
                <span className="text-3xl align-top">%</span>
              </div>
              <div className="text-sm uppercase text-grey tracking-wide">
                Progress Of The Last Book
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className="flex-mx-4">
        <div className="w-full mb-6 lg:mb-0 flex flex-col">
          <div className="flex-grow flex flex-col bg-white border-t border-b sm:rounded sm:border shadow overflow-hidden">
            <div className="border-b">
              <div className="flex justify-between px-6 -mb-px">
                <h3 className="text-blue-dark py-4 font-normal text-lg">
                  Books
                </h3>
                <div className="flex items-center">
                  <button className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded">
                    Add New
                  </button>
                </div>
              </div>
            </div>
            <div className="flex-grow flex px-6 py-6 text-grey-darker items-center -mx-4">
              <div className="w-2/5 xl:w-1/4 px-4 flex items-center">
                <span className="text-lg">22 Char</span>
              </div>
              <div className="hidden md:flex lg:hidden xl:flex w-1/4 px-4 items-center justify-between">
                <div className="bg-yellow-400 h-2 w-48 rounded-full mr-2"></div>
                100%
              </div>
              <div className="flex w-3/5 md:w/12 items-center">
                <div className="w-1/2 px-4">
                  <p className="text-right">100/100 Pages</p>
                </div>
                <div className="w-1/2 px-4 flex justify-end">
                  <button className="bg-blue-500 hover:bg-blue-700 text-white py-2 px-2 mr-2 rounded">
                    <MdVisibility size={20} />
                  </button>
                  <button className="bg-orange-400 hover:bg-orange-600 text-white px-2 py-2 mr-2 rounded">
                    <MdEdit size={20} />
                  </button>
                  <button className="bg-red-500 hover:bg-red-700 text-white px-2 py-2 rounded">
                    <MdDelete size={20} />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Home;
