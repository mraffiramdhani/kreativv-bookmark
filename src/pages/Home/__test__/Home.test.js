import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Home from "../index";

afterEach(cleanup);

describe("Examine Home Component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
