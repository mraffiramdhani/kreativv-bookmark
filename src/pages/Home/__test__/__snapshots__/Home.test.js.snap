// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Examine Home Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="font-sans antialiased h-screen flex"
  >
    <div
      class="bg-gray-800 text-purple-300 w-24 hidden md:block"
    >
      <div
        class="flex flex-col"
      >
        <div
          class="h-12 my-4"
        >
          <p
            class="text-gray-500 text-center text-4xl"
          >
            KB
          </p>
        </div>
        <a
          class="flex flex-col text-gray-600 hover:text-gray-400 items-center my-8"
          href="/home"
        >
          <svg
            fill="currentColor"
            height="34"
            size="34"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="34"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
            />
          </svg>
          <p
            class="text-sm"
          >
            Dashboard
          </p>
        </a>
        <a
          class="flex flex-col text-gray-600 hover:text-gray-400 items-center my-8"
          href="/account"
        >
          <svg
            fill="currentColor"
            height="34"
            size="34"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="34"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 5v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2zm12 4c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zm-9 8c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6v-1z"
            />
          </svg>
          <p
            class="text-sm"
          >
            Profile
          </p>
        </a>
        <button
          class="flex flex-col text-gray-600 hover:text-gray-400 items-center my-8"
        >
          <svg
            fill="currentColor"
            height="34"
            size="34"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="34"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13 3h-2v10h2V3zm4.83 2.17l-1.42 1.42C17.99 7.86 19 9.81 19 12c0 3.87-3.13 7-7 7s-7-3.13-7-7c0-2.19 1.01-4.14 2.58-5.42L6.17 5.17C4.23 6.82 3 9.26 3 12c0 4.97 4.03 9 9 9s9-4.03 9-9c0-2.74-1.23-5.18-3.17-6.83z"
            />
          </svg>
          <p
            class="text-sm"
          >
            Log Out
          </p>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
